### PR TITLE
Update location.yaml

### PR DESCRIPTION
**Pull request by ukkaps**
Included request parameter “MaxLocationAge’ to give the developer option to mention the maximum aging of the location info, accepted by the application. 

If the developer wants only real-time or near real-time information, we as operators then can avoid validating against stale location info resulted from network reasons such as device switched-off, not in servicing area, device issues etc.

### CODE REVIEW

File: code/API_definitions/location.yaml [hunk 1]
```diff
@@ -97,11 +97,14 @@ components:
...
         accuracy:
           $ref: '#/components/schemas/Accuracy'
+        maxLocationAge:
+          $ref: '#/components/schemas/MaxLocationAge'
       required:
         - ueId
         - latitude
         - longitude
         - accuracy
+        - maxLocationAge
```
Comment (jlurien) [inline code/API_definitions/location.yaml:107]: Maybe it should be optional, with a default value. It depends on how often it is expected to be used. If the common case is to request "real-time" info (less than a minute or few minutes), it is not convenient to ask for it in every request.

**Review by jlurien — COMMENTED** [review]
It is a good addition as long as it is expected that some implementations may rely on stale data. In TEF we were thinking on a "real-time" implementation, so this input is not really required, but I would like to hear opinions from other operators. If convenient, we think it would be better to leave it optional with a default value to the minimum. We also need to define the expected behaviour when the location is known but the age threshold is not met. Should API return a response with a warning, an error, a false response...?

**Comment by jlurien** [comment]
We also have to review the versioning process. Any modification should increase current version (0.1.0). We may need to create a new development branch towards 0.2.0

**Comment by bigludo7** [comment]
Hello
This is a valuable request. We need on our side a little bit more time to explore 3GPP api capabilities (which support the camara api implementation on our side). 

Agreed on @jlurien comments on versioning - as we have already begun implementation it's better to change swagger version

**Comment by patrice-conil** [comment]
Hello @ukkaps @jlurien ,
As MaxLocationAge = 0  can  mean for "current" or "real time" ... I vote for mandatory version.

Patrice

**Comment by ukkaps** [comment]
@bigludo7  @patrice-conil  @jlurien 
API version updated as suggested from 0.1.0 to 0.2.0

File: code/API_definitions/location.yaml (hunk 1 above)
Comment (patrice-conil) [reply code/API_definitions/location.yaml:107]: A default value for maxLocationAge is a good idea. I vote for.

File: code/API_definitions/location.yaml [hunk 2]
```diff
@@ -181,6 +184,12 @@ components:
       minimum: 2
       maximum: 200
       example: 50
+    MaxLocationAge:
+      description: Max location aging validity required by the application in Minute
+      type: number
+      minimum: 1
+      maximum: 1440
+      example: 30
```
Comment (patrice-conil) [inline code/API_definitions/location.yaml:192]: default: 0

**Review by patrice-conil — CHANGES_REQUESTED** [review]
Concerning Ipv6Addr. 
As the second pattern is much too permissive and does not add anything more, I suggest deleting it.

    Ipv6Addr:
      type: string
      format: ipv6
      pattern: '^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))(\/(([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))?$'
      example: '2001:db8:85a3:8d3:1319:8a2e:370:7344'

File: code/API_definitions/location.yaml (hunk 1 above)
Comment (jlurien) [inline code/API_definitions/location.yaml:107]:
```suggestion
        - maxAge 
```
Location is implicit in the context

File: code/API_definitions/location.yaml (hunk 2 above)
Comment (jlurien) [reply code/API_definitions/location.yaml:192]: I have to check the technical implications, but I assume that maxAge=0 implies that implementation must not rely on any kind of stale or cached data. If we set the default to this, we are forcing that requests which not include explicitly this parameter would get an error, if some implementation is caching some data for few minutes, which could not be the intention of the consumer.

**Comment by jlurien** [comment]
Hi, in order to manage versioning and keep current agreed version in main, I have copy what we are doing in QualityOnDemand WG, and I have created a new branch **dev-0.2.0** where we should propose any new PR for 0.2.0, so please @ukkaps could you switch your proposal to be merged in the new branch instead of main.

**Review by jlurien — COMMENTED** [review]
Some comments on the behaviour and implications of the default value. Also, please, could you move this PR to the new dev-0.2.0 branch. We should have created the new branch in advance but it's better now that when new changes arrive.

**Comment by ukkaps** [comment]
@jlurien , new PR is raised [here](https://github.com/camaraproject/DeviceLocation/pull/9) as suggested above. May I close the old PR (this) now?

**Comment by MarkusKuemmerle** [comment]
> @jlurien , new PR is raised [here](https://github.com/camaraproject/DeviceLocation/pull/9) as suggested above. May I close the old PR (this) now?
> 
> @MarkusKuemmerle , do we have any timelines to release the code of this for MWC implementation?



> @jlurien , new PR is raised [here](https://github.com/camaraproject/DeviceLocation/pull/9) as suggested above. May I close the old PR (this) now?
> 
> @MarkusKuemmerle , do we have any timelines to release the code of this for MWC implementation?

The 0.1.0 version which was merged in main branch in November will be the version for the MWC demos. Nevertheless it's good to further improve the API and to work on further versions.

**Comment by jlurien** [comment]
> @jlurien , new PR is raised [here](https://github.com/camaraproject/DeviceLocation/pull/9) as suggested above. May I close the old PR (this) now?
> 
> @MarkusKuemmerle , do we have any timelines to release the code of this for MWC implementation?

@ukkaps, yes, you can close this now and we can follow discussions in the new one. Thanks!

**Pull request closed without merging**